Src/ILGPU.Tests/Generic/TestBase.cs

```diff
@@ -106,6 +106,7 @@ protected TestBase(ITestOutputHelper output, TestContext testContext)
             params object[] arguments)
             where TIndex : struct, IIndex
         {
+            Accelerator.Synchronize();
```

Comment:
I added this synchronization point to avoid potential problems related to other operations that use the default accelerator current. These have synchronized before running a test kernel using a different stream 🤓